api/app.py

```diff
@@ -166,32 +142,29 @@ def update_recipe(recipe_id):
         name = category.get('name')
         color = category.get('color', '#808080')
         cat = Category.query.filter_by(name=name).first()
-        if not cat:
-            cat = Category(name=name, color=color)
-            db.session.add(cat)
-        categories.append(cat)
+        if cat:
+            categories.append(cat)
+        else:
+            return jsonify({'error': f'Category {name} doesnt exist'}), 400
 
     for ing in recipe.ingredients:
         db.session.delete(ing)
     db.session.flush()
 
     ingredients = []
-    ingredient_regex = re.compile(r'^(?P<quantity>\d+)(?P<unit>[a-zA-Z]*) (?P<name>.+)$')
     for ingredient in data.get('ingredients', []):
-        match = ingredient_regex.match(ingredient.strip())
-        if not match:
-            return jsonify({'error': f'Invalid ingredient format: {ingredient}'}), 400
         ingredient = Ingredient(
-            name=match['name'],
-            unit=match['unit'] or None,
-            quantity=float(match['quantity']),
+            name=ingredient['name'],
+            unit=ingredient['unit'],
+            quantity=float(ingredient['quantity']),
             recipe=recipe
         )
         ingredients.append(ingredient)
 
     recipe.name = data.get('name')
     recipe.duration = data.get('duration')
     recipe.pictures = data.get('pictures')
+    recipe.pictures = ",".join(data.get('pictures', []))
```

Comment:
We receive `pictures` as a list of URLs from frontend, so there's no need for splitting:
```json
"pictures": [
    "https://staticcookist.akamaized.net/wp-content/uploads/sites/22/2024/09/THUMB-VIDEO-2_rev1-56.jpeg"
  ]
```